challenge_9/scala/ndsmith3/Squares.scala

```diff
@@ -0,0 +1,38 @@
+object Squares {
+  // Lazy values are used in the sortSquares function to avoid exceptions from
```

Comment:
This solution is pretty convoluted.  Yo could just use 2 indexes to work your way from the outsides in on the source array, copying to the target array.